### PR TITLE
docs: change to sphinx docs

### PR DESCRIPTION
**Pull request by kemingy**
## Pros

* looks better for the API interface part
* Python ecosystem

## Cons

* the doc of sphinx and myst_parser is not great

### CODE REVIEW

**Comment by kemingy** [comment]
TODO
- docstrings are not compatible with sphinx

You can build the doc site with `make doc` locally.

**Comment by kemingy** [comment]
> TODO
> 
>     * docstrings are not compatible with sphinx
> 

- Refer to https://github.com/executablebooks/MyST-Parser/issues/228

We can add some patches to support Markdown in docstrings. It might be possible to officially support Markdown in the future.

**Comment by kemingy** [comment]
> > TODO
> > ```
> > * docstrings are not compatible with sphinx
> > ```
> 
>     * Refer to [Allow for autodoc to parse Markdown docstrings executablebooks/MyST-Parser#228](https://github.com/executablebooks/MyST-Parser/issues/228)
> 
> 
> We can add some patches to support Markdown in docstrings. It might be possible to officially support Markdown in the future.

Let's use the default reStructuredText format for now. :cry:

File: mosec/worker.py [hunk 1]
```diff
@@ -72,12 +65,20 @@ def __init__(self):
         This method doesn't require the child class to override.
         """
 
-    def serialize_ipc(self, data) -> bytes:
-        """Define IPC serialize method."""
+    def serialize_ipc(self, data: Any) -> bytes:
+        """Define IPC serialization method.
+
+        Args:
+            data: returned data :py:meth:`forward`
```
Comment (lkevinzc) [inline mosec/worker.py:72]:
```suggestion
            data: returned data from :py:meth:`forward`
```

**Review by lkevinzc — COMMENTED** [review]
I think we can temporarily remove the DEV GUIDE/Design section.

**Comment by kemingy** [comment]
> I think we can temporarily remove the DEV GUIDE/Design section.

Removed.

**Review by lkevinzc — APPROVED** [review]

**Pull request merged**